operator/standalone/delete.go

```diff
@@ -0,0 +1,104 @@
+package standalone
+
+import (
+	"context"
+	pipeline "github.com/ccremer/go-command-pipeline"
+	helmv1beta1 "github.com/crossplane-contrib/provider-helm/apis/release/v1beta1"
+	"github.com/vshn/appcat-service-postgresql/apis/postgresql/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+// DeleteStandalonePipeline is a pipeline that deletes an instance from the target deployment namespace.
+type DeleteStandalonePipeline struct {
+	client             client.Client
+	instance           *v1alpha1.PostgresqlStandalone
+	helmReleaseDeleted bool
+}
+
+// NewDeleteStandalonePipeline creates a new delete pipeline with the required dependencies.
+func NewDeleteStandalonePipeline(client client.Client, instance *v1alpha1.PostgresqlStandalone) *DeleteStandalonePipeline {
+	return &DeleteStandalonePipeline{
+		instance: instance,
+		client:   client,
+	}
+}
+
+// RunPipeline executes the pipeline with configured business logic steps.
+// The pipeline requires multiple reconciliations due to asynchronous deletion of resources in background
+// The Helm Release step requires a complete removal of its resources before moving to the next step
+func (d *DeleteStandalonePipeline) RunPipeline(ctx context.Context) error {
+	return pipeline.NewPipeline().
+		WithSteps(
+			pipeline.NewStepFromFunc("delete connection secret", d.deleteConnectionSecret),
+			pipeline.NewStepFromFunc("delete helm release", d.deleteHelmRelease),
+			pipeline.If(d.isHelmReleaseDeleted,
+				pipeline.NewPipeline().WithNestedSteps("finalize",
+					pipeline.NewStepFromFunc("delete namespace", d.deleteNamespace),
+					pipeline.NewStepFromFunc("remove finalizer", d.removeFinalizer),
+				),
+			),
+		).
+		RunWithContext(ctx).Err()
+}
+
+// deleteHelmRelease removes the Helm Release from the cluster
+// We may reconcile multiple times until Release is completely deleted hence we use helmReleaseDeleted variable
+func (d *DeleteStandalonePipeline) deleteHelmRelease(ctx context.Context) error {
+	helmRelease := &helmv1beta1.Release{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      d.instance.Status.HelmChart.DeploymentNamespace,
+			Namespace: d.instance.Status.HelmChart.DeploymentNamespace,
+		},
+	}
+	err := d.client.Delete(ctx, helmRelease)
+	if err != nil && apierrors.IsNotFound(err) {
+		d.helmReleaseDeleted = true
+	}
+	return skipOnNotFound(err)
```

Comment:
Actually, I've found an already implemented function in the controller-runtime library 😄 
(to be applied in other functions as well)
```suggestion
	return client.IgnoreNotFound(err)
```